python/prairielearn.py

```diff
@@ -1776,3 +1776,28 @@ def index2key(i):
         key = chr(ord("a") + i)
 
     return key
+
+
+def is_within_limits(n: int) -> bool:
+    return -((2**53) - 1) <= n <= 2**53 - 1
+
+
+def all_integers_within_limits(item: Any) -> None:
+    """Raise an exception if the input item contains any oversized integers."""
+    item_stack = [item]
+
+    while item_stack:
+        next_item = item_stack.pop()
+
+        if isinstance(next_item, int):
+            if not is_within_limits(next_item):
+                raise ValueError(
+                    f"Data structure contains oversized integer: {next_item}"
+                )
+
+        elif isinstance(next_item, list):
+            item_stack.extend(next_item)
+
+        elif isinstance(next_item, dict):
+            item_stack.extend(next_item.keys())
+            item_stack.extend(next_item.values())
```

Comment:
IMO we should inline these into the zygote (or maybe `zygote_utils.py` so that we can import it in tests); I don't want these to be part of our public API if we can avoid it.

Reply:
Good call! I think having a `zygote_utils.py` makes the most sense.

Reply:
Moved here: https://github.com/PrairieLearn/PrairieLearn/pull/6980/commits/164b0ac219e30f8bcc2eb5d1730b4aabdf792c90